src/OpenTelemetry.Api/Trace/SpanHelper.cs

```diff
@@ -121,48 +83,38 @@ public static Status ResolveSpanStatusForHttpStatusCode(int httpStatusCode)
         /// <returns>Resolved span <see cref="Status"/> for the Grpc status code.</returns>
         public static Status ResolveSpanStatusForGrpcStatusCode(int statusCode)
         {
-            var newStatus = Status.Unknown;
+            var newStatus = Status.Unset;
 
             if (typeof(StatusCanonicalCode).IsEnumDefined(statusCode))
             {
-                newStatus = new Status((StatusCanonicalCode)statusCode);
+                newStatus = ((StatusCanonicalCode)statusCode) switch
+                {
+                    StatusCanonicalCode.Ok => Status.Ok,
+                    _ => Status.Error,
+                };
             }
 
             return newStatus;
         }
 
         /// <summary>
-        /// Helper method that returns Status from <see cref="StatusCanonicalCode"/> to save on allocations.
+        /// Helper method that returns Status from <see cref="StatusCode"/> to save on allocations.
         /// </summary>
-        /// <param name="statusCanonicalCode"><see cref="StatusCanonicalCode"/>.</param>
-        /// <returns>Resolved span <see cref="Status"/> for the Canonical status code.</returns>
-        public static Status ResolveCanonicalCodeToStatus(string statusCanonicalCode)
+        /// <param name="statusCode"><see cref="StatusCode"/>.</param>
+        /// <returns>Resolved span <see cref="Status"/> for the status code.</returns>
+        public static Status ResolveStatusCodeToStatus(string statusCode)
```

Comment:
Why do we need this?

Reply:
This is used to get the Status from Activity.

```csharp
public static Status GetStatus(this Activity activity)
        {
            Debug.Assert(activity != null, "Activity should not be null");

            ActivityStatusTagEnumerator state = default;

            ActivityTagsEnumeratorFactory<ActivityStatusTagEnumerator>.Enumerate(activity, ref state);

            var status = SpanHelper.ResolveStatusCodeToStatus(state.StatusCode);

            if (status.IsValid && !string.IsNullOrEmpty(state.StatusDescription))
            {
                return status.WithDescription(state.StatusDescription);
            }

            return status;
        }
```

Reply:
I think the code from Activity should be an integer, and here what we need to do is to retrieve the integer and cast it to the status enum (and have a default behavior in case cast failed).
String parsing should be avoided.

Reply:
I see your point. From spec side, it's not clear if we should save the value as enum or string:
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#set-status

Reply:
I guess it is clear that `otel.status_code` should be int and `otel.status_description` should be string.